packages/cli-lib/lib/handleFieldsJs.js

```diff
@@ -0,0 +1,94 @@
+const fs = require('fs');
+const path = require('path');
+const { logger } = require('../logger');
+const { i18n } = require('@hubspot/cli-lib/lib/lang');
+const { getExt } = require('../path');
+const i18nKey = 'cli.commands.upload';
+
+function handleFieldErrors(e, filePath) {
+  if (e instanceof SyntaxError) {
+    const ext = getExt(filePath);
+    if (ext === 'json') {
+      logger.error(
+        i18n(`${i18nKey}.errors.jsonParsingFailed`, { json: filePath })
+      );
+    } else if (ext == 'js') {
+      logger.error(i18n(`${i18nKey}.errors.jsSyntaxError`, { js: filePath }));
+    }
+  }
+  if (e.code === 'ENOENT') {
+    logger.error(
+      i18n(`${i18nKey}.errors.invalidPath`, {
+        path: filePath,
+      })
+    );
+  }
+  if (e.code === 'MODULE_NOT_FOUND') {
+    logger.error(
+      i18n(`${i18nKey}.errors.jsSyntaxError`, {
+        path: filePath,
+      })
+    );
+  }
+}
+
+/**
+ * Converts a fields.js file into a fields.json file, writes, and returns of fields.json
+ * @param {string} file - The path of the fields.js javascript file.
+ * @param {string[]} options - Optional arguments to pass to the exported function in fields.js
+ * @returns {string} The path of the written fields.json file.
+ */
+async function convertFieldsJs(filePath, options) {
+  const dirName = path.dirname(filePath);
+  logger.info(
+    i18n(`${i18nKey}.converting`, {
+      src: dirName + '/fields.js',
+      dest: dirName + '/fields.json',
+    })
+  );
+
+  try {
+    // If we do not clear the cache, hs watch will not work
+    delete require.cache[filePath];
```

Comment:
What issue were you seeing with `hs watch`?

Reply:
Since the fields.js file is called with `require`, node will cache it like a module. So when the fields.js is updated and it's recompiled, when it hits the `require()` line, node returns the cached version of the fields.js file rather than the updated version. See https://nodejs.org/api/modules.html#caching

Reply:
Ahhh, makes sense! It might be worth rewording this comment since it _sounds like_ `hs watch` does work correctly but the cache might trip up `convertFieldsJs`